api/envoy/extensions/filters/http/on_demand/v3/on_demand.proto

```diff
@@ -11,11 +15,47 @@ option java_multiple_files = true;
 option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/http/on_demand/v3;on_demandv3";
 option (udpa.annotations.file_status).package_version_status = ACTIVE;
 
-// [#protodoc-title: OnDemand]
-// IP tagging :ref:`configuration overview <config_http_filters_on_demand>`.
+// [#protodoc-title: On Demand Discovery]
+// On Demand Discovery :ref:`configuration overview <config_http_filters_on_demand>`.
 // [#extension: envoy.filters.http.on_demand]
 
+// On Demand Discovery filter config.
 message OnDemand {
   option (udpa.annotations.versioning).previous_message_type =
       "envoy.config.filter.http.on_demand.v2.OnDemand";
+
+  // An optional configuration for on-demand cluster discovery
+  // service. If not specified, the on-demand cluster discovery will
+  // be disabled. When it's specified, the filter will pause a request
+  // to an unknown cluster and will begin a cluster discovery
+  // process. When the discovery is finished (successfully or not),
+  // the request will be resumed for further processing.
+  config.core.v3.ConfigSource odcds_config = 1;
```

Comment:
Sorry I didn't point this out before, but I'm thinking that we should put this in an OnDemandCds submessage, since it's really multiple related fields.

Reply:
No problem, done.